lib/src/linting/rules/no-objects-in-query-params.ts

```diff
@@ -0,0 +1,22 @@
+import { flatten } from "lodash";
+import { LintingRule } from "../rule";
+
+import { TypeNode } from "../../models/nodes";
+import { extractQueryParams } from "../../utilities/extract-endpoint-types";
+import { extractNestedObjectTypes } from "../../utilities/extract-nested-types";
+
+/**
+ * Ensure query parameters are not objects
+ */
+export const noObjectsInQueryParams: LintingRule = contract => {
+  const extractObjectTypes = (t: TypeNode) =>
+    t && extractNestedObjectTypes(t, contract.types);
```

Comment:
Is `t &&` actually needed?

```suggestion
    extractNestedObjectTypes(t, contract.types);
```